src/client/Microsoft.Identity.Client/Cache/Items/MsalIdTokenCacheItem.cs

```diff
@@ -53,13 +53,9 @@ internal MsalIdTokenCacheItem()
         }
 
     
-        internal bool IsAdfs { get; set; }
+        internal bool IsAdfs { get; set; } // TODO: get rid of this crap
```

Comment:
>// TODO: get rid of this crap [](http://example.com/codeflow?start=43&length=29)

? what does this comment suggest? pls. be more specific.

Reply:
Will do :). This was not added by our team, btw. It's wrong because when you load a token from the cache, you don't know the authority so `IsAdfs` is always false. It changes how we report UPN and a few other things.